### PR TITLE
Invert colors of pillboxes in error report

### PR DESCRIPTION
**Pull request by pr33thi**
[https://dimagi-dev.atlassian.net/projects/HI/issues/HI-375?filter=myopenissues](Interrupt Ticket)

Probably easiest to review commit by commit. This used some code that was written a long time ago, so I refactored out a class that was mostly unused.

@orangejenny @czue

### CODE REVIEW

File: corehq/apps/reports/standard/forms/filters.py [hunk 1]
```diff
@@ -21,15 +21,16 @@ class SubmissionErrorType(object):
...
     human_readable = [ugettext_lazy("Normal Form"),
                       ugettext_lazy("Form with Errors"),
                       ugettext_lazy("Duplicate Form"),
                       ugettext_lazy("Overwritten Form"),
                       ugettext_lazy("Generic Error"),
                       ugettext_lazy("Archived Form")]
-    
+
     error_defaults = [False, True, False, False, True, False]
     success_defaults = [True, False, False, False, False, False]
```
Comment (orangejenny) [inline corehq/apps/reports/standard/forms/filters.py]: Tangential to this PR, but it looks like `success_defaults` isn't referenced anywhere and can be removed.

File: corehq/apps/reports/standard/forms/filters.py [hunk 2]
```diff
@@ -74,7 +75,4 @@ def filter_context(self):
     @classmethod
     def get_filter_toggle(cls, request):
         filter_ = request.GET.getlist(cls.slug, None)
-        if filter_:
-            return SubmissionErrorType.use_filter(filter_)
-        else:
-            return SubmissionErrorType.use_error_defaults()
+        return SubmissionErrorType.use_filter(filter_)
```
Comment (orangejenny) [inline corehq/apps/reports/standard/forms/filters.py]: This is getting rid of the default selected options, isn't it? Why?

File: corehq/apps/reports/static/reports/js/filters/button_group.js [hunk 3]
```diff
@@ -20,16 +20,6 @@ hqDefine("reports/js/filters/button_group", ['jquery'], function ($) {
             }
             $activeCheckbox.trigger('change');
 
-            if ((!$el.children().hasClass('btn-primary')) && !canBeEmpty) {
```
Comment (orangejenny) [inline corehq/apps/reports/static/reports/js/filters/button_group.js]: `canBeEmpty` can be removed from the param list - I'm surprised the linter didn't flag this.

File: corehq/apps/reports/templates/reports/filters/submit_error_types.html [hunk 4]
```diff
@@ -4,7 +4,7 @@
 {% block filter_content %}
     <div id="{{ css_id }}" class="btn-group btn-group-separated report-filter-button-group" data-toggle="buttons-checkbox">
     {% for type in submission_types %}
-        <button class="btn btn-default{% if type.show %} btn-default{% else %} active{% endif %}" data-checkfilter="submission-filter-{{ forloop.counter }}">{{ type.name }}</button>
```
Comment (orangejenny) [inline corehq/apps/reports/templates/reports/filters/submit_error_types.html]: There's similar logic in https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/templates/reports/filters/filter_users.html#L13 which also uses the `report-filter-button-group` class, does that template also need to be updated?

File: corehq/apps/reports/standard/forms/filters.py [hunk 5]
```diff
@@ -13,59 +13,21 @@ def __init__(self,  type, show, name, doc_type):
...
 
-class SubmissionErrorType(object):
-    # This is largely modeled off how the user filter works
-    SUCCESS = 0
-    PARTIAL_SUCCESS = 1
-    DUPLICATE = 2
-    OVERWRITTEN = 3
-    UNKNOWN_ERROR = 4
-    ARCHIVED = 5
+class SubmissionTypeFilter(BaseReportFilter):
```
Comment (orangejenny) [inline corehq/apps/reports/standard/forms/filters.py]: Nice cleanup.

**Review by orangejenny — CHANGES_REQUESTED** [review]
Minor feedback.

File: corehq/apps/reports/standard/forms/filters.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/reports/standard/forms/filters.py]: Nevermind, I see now you took care of this.

**Comment by czue** [comment]
looks like jenny already gave a better review than i could

File: corehq/apps/reports/standard/forms/filters.py (hunk 2 above)
Comment (pr33thi) [reply corehq/apps/reports/standard/forms/filters.py]: The defaults looked arbitrary (it had the `Form with Errors` and `Generic Error` buttons default to being selected), I figured it was legacy code that didn't make sense anymore, so now it just default to every button being deselected

Comment (orangejenny) [reply corehq/apps/reports/standard/forms/filters.py]: @djmore9 should there be a default value? We're talking about the "Submission Type" filter on the [Raw Forms, Errors & Duplicates report](https://www.commcarehq.org/a/bosco/reports/submit_errors/)

File: corehq/apps/reports/templates/reports/filters/submit_error_types.html (hunk 4 above)
Comment (pr33thi) [reply corehq/apps/reports/templates/reports/filters/submit_error_types.html]: How do I actually view that page on hq? I can trace the html back to this url: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/urls.py#L130, but it seems to be the url for the api?

File: corehq/apps/reports/templates/reports/filters/submit_error_types.html [hunk 6]
```diff
@@ -4,7 +4,7 @@
...
+        <button class="{% if type.show %} active{% else %} btn-default{% endif %}" data-checkfilter="submission-filter-{{ forloop.counter }}">{{ type.name }}</button>
```
[the remark this answers is not in the document]
Comment (pr33thi) [reply corehq/apps/reports/templates/reports/filters/submit_error_types.html]: https://github.com/dimagi/commcare-hq/pull/23238/commits/016178d497704962133d6e78f6b90130bd3f6d78

File: corehq/apps/reports/static/reports/js/filters/button_group.js (hunk 3 above)
Comment (pr33thi) [reply corehq/apps/reports/static/reports/js/filters/button_group.js]: https://github.com/dimagi/commcare-hq/pull/23238/commits/016178d497704962133d6e78f6b90130bd3f6d78

File: corehq/apps/reports/templates/reports/filters/submit_error_types.html (hunk 4 above)
Comment (orangejenny) [reply corehq/apps/reports/templates/reports/filters/submit_error_types.html]: I think you're right that `UserTypeFilter` is never visible in main HQ, but it's used in custom reports: https://www.commcarehq.org/a/pact/reports/custom/chws/ And it does look like that report uses the same inverted behavior, where white means selected and purple means not selected.

You can't test custom reports locally. I'd just make the same changes to filter_user.html that you made to this file and leave yourself a note to test that report post-deploy.

Comment (pr33thi) [reply corehq/apps/reports/templates/reports/filters/submit_error_types.html]: Sounds good, https://github.com/dimagi/commcare-hq/pull/23238/commits/43e5ac21742711c01673176214fdadd5d698215f
And I made a note to myself to check this after tomorrow's deploy

**Review by orangejenny — APPROVED** [review]

**Pull request merged**